streams/src/test/java/org/apache/kafka/streams/kstream/internals/WindowedStreamPartitionerTest.java

```diff
@@ -138,10 +138,88 @@ public void testWindowedDeserializerNoArgConstructors() {
         props.remove("value.deserializer.inner.class");
         WindowedDeserializer<?> windowedDeserializer1 = new WindowedDeserializer<>();
         windowedDeserializer1.configure(props, false);
-        Deserializer<?> inner1 = windowedDeserializer1.innerDeserializer();
+        final Deserializer<?> inner1 = windowedDeserializer1.innerDeserializer();
         assertNotNull("Inner deserializer should be not null", inner1);
         assertTrue("Inner deserializer type should be ByteArrayDeserializer", inner1 instanceof ByteArrayDeserializer);
         windowedDeserializer.close();
         windowedDeserializer1.close();
     }
+    
+    @Test
+    public void testWindowSerializeConfig() {
+        final Map<String, String> props = new HashMap<>();
+        // test key[value].serializer.inner.class takes precedence over serializer.inner.class
+        final WindowedDeserializer<StringSerializer> windowedDeserializer = new WindowedDeserializer<>();
+        props.put("key.deserializer.inner.class", "org.apache.kafka.common.serialization.StringDeserializer");
+        props.put("deserializer.inner.class", "org.apache.kafka.common.serialization.StringDeserializer");
+        windowedDeserializer.configure(props, true);
+        final Deserializer<?> inner = windowedDeserializer.innerDeserializer();
+        assertTrue("Inner deserializer type should be StringDeserializer", inner instanceof StringDeserializer);
+    }
+    
+    @Test
+    public void testWindowSerializeWithWindowSize() {
+        final Map<String, String> props = new HashMap<>();
+        // test key[value].serializer.inner.class takes precedence over serializer.inner.class
+        final WindowedDeserializer<StringSerializer> windowedDeserializer = new WindowedDeserializer<>();
+        props.put("key.deserializer.inner.class", "org.apache.kafka.common.serialization.StringDeserializer");
+        props.put("deserializer.inner.class", "org.apache.kafka.common.serialization.StringDeserializer");
+        windowedDeserializer.configure(props, true);
+        props.put("deserializer.inner.class", "org.apache.kafka.common.serialization.ByteArraySerializer");
+        props.remove("key.serializer.inner.class");
+        props.remove("value.serializer.inner.class");
+        //test for serializer expected window end time
```

Comment:
seems actual test start here? if yes, remove everything from above. It seems test name should be `shouldReturnWindowWithCorrectSizeWhenDeserializing` ?

Reply:
The deserializer.inner.class in props map could not be removed, or otherwise that will result in test failure. 
Description of the failure:
```
org.apache.kafka.streams.kstream.internals.WindowedStreamPartitionerTest > testWindowSerializeWithWindowSize FAILED
    java.lang.ClassCastException: class org.apache.kafka.common.serialization.ByteArraySerializer
        at java.lang.Class.asSubclass(Class.java:3404)
        at org.apache.kafka.common.utils.Utils.newInstance(Utils.java:308)
        at org.apache.kafka.streams.kstream.internals.WindowedDeserializer.configure(WindowedDeserializer.java:66)
        at org.apache.kafka.streams.kstream.internals.WindowedStreamPartitionerTest.testWindowSerializeWithWindowSize(WindowedStreamPartitionerTest.java:166) ```

Reply:
Makes sense. There is still some cleanup to do.... `windowedDeserializer` is not use anymore in the test but only `windowedDeserializer1` -- also, setting up `props` has `put` and later `remove` that can both be removed...

When cleaning up: consider what is *required* to set up the test, and what is not. Remove everything that is not required.

Reply:
Removing the properties will result in the following exception: 

'''
org.apache.kafka.streams.kstream.internals.WindowedStreamPartitionerTest > testWindowSerializeExpectedWindowSize FAILED
    java.lang.NullPointerException
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:348)
        at org.apache.kafka.common.utils.Utils.newInstance(Utils.java:308)
        at org.apache.kafka.streams.kstream.internals.WindowedDeserializer.configure(WindowedDeserializer.java:67)
        at org.apache.kafka.streams.kstream.internals.WindowedStreamPartitionerTest.testWindowSerializeExpectedWindowSize(WindowedStreamPartitionerTest.java:154)
'''

In other words, the arrangement adopted from the first test method could not be changed.

Reply:
Also, the windowedDeserializer's inner class is needed for instantiation in windowedDeserializer1. It could not be removed.